File: Lab1/lab1.py
from math import sqrt

#input
a = float(input("Введіть а: "))
b = float(input("Введіть b: "))
c = float(input("Введіть c: "))

def validate_triangle(a, b, c):
    if any(side <= 0 for side in [a,b,c]):
        raise Exception("Вибачте, але сторони трикутника повинні бути більшими від нуля") 

    a, b, c = sorted([a, b, c])
    if a + b <= c:
       raise Exception("Вибачте, але трикутник існує тоді і тільки тоді, коли сума будь-яких двох його сторін більше третьою боку") 

#a, b are sides of the angle bisector divides
#side_against_bisector is the side bisector is drawn to
def bisector(a, b, side_against_bisector):
    c = side_against_bisector
    return sqrt(a * b * (a + b + c) * (a + b - c)) / (a + b)

validate_triangle(a,b,c)

bisector_a = bisector(c, b, a)
print("Бісектриса проведена до а: {:.2f}".format(bisector_a))

bisector_b = bisector(a, c, b)
print("Бісектриса проведена до b: {:.2f}".format(bisector_b))

bisector_c = bisector(a, b, c)
print("Бісектриса проведена до c: {:.2f}".format(bisector_c))

# evaluate radiuses 
# calculate the area  
semiPerimeter = (a + b + c) / 2 
area = sqrt(semiPerimeter*(semiPerimeter-a)*(semiPerimeter-b)*(semiPerimeter-c))

R = (a * b * c) / (4 * area)
print("Радіус описаного кола: {:.2f}".format(R))

r = area / semiPerimeter
print("Радіус вписаного кола: {:.2f}".format(r))

File: Lab1/lab1.cpp
#include <iostream>
#include <math.h>
using namespace std;


float calcBisector(float a, float b,float sideAgainstBisector){
    float c = sideAgainstBisector;
    float result = sqrt(a * b * (a + b + c) * (a + b - c)) / (a + b);
    return result;
}

void validateTriangle(float a, float b,float c){
    if (a <= 0 || b <= 0 || c <= 0) {
        throw "Вибачте, але сторони трикутника повинні бути більшими від нуля";
    }
    if (a + b <= c || a + c <= b || c + b <= a){
        throw "Вибачте, але трикутник існує тоді і тільки тоді, коли сума будь-яких двох його сторін більше третьою боку";
    }
}

int main()
{
    float a;
    float b;
    float c;

    cout << "Введіть а: ";
    cin >> a;

    cout << "Введіть b: ";
    cin >> b;

    cout << "Введіть c: ";
    cin >> c;

    float bisectorA = calcBisector(c, b, a);
    float bisectorB = calcBisector(a, c, b);
    float bisectorC = calcBisector(a, b, c);

    cout << "Бісектриса проведена до a: " << bisectorA << "\n";
    cout << "Бісектриса проведена до b: " << bisectorB << "\n";
    cout << "Бісектриса проведена до c: " << bisectorC << "\n";
    
    float semiPerimeter = (a + b + c) / 2;
    float area = sqrt(semiPerimeter*(semiPerimeter-a)*(semiPerimeter-b)*(semiPerimeter-c));

    float R = (a * b * c) / (4 * area);
    cout << "Радіус описаного кола: " << R << "\n";

    float r = area / semiPerimeter;
    cout << "Радіус вписаного кола: " << r << "\n";

    return 0;
}
